core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -128,7 +129,24 @@ private[client] class CachedEventHubsReceiver private (ehConf: EventHubsConf,
   }
 
   private def closeReceiver(): Future[Void] = {
-    retryJava(receiver.close(), "closing a receiver")
+    // Closing a PartitionReceiver is not a retryable operation: after the first call, it always
+    // returns the same CompletableFuture. Therefore, try once. If it fails with a transient
+    // error, log and ignore.
+    val taskId = EventHubsUtils.getTaskId
+    toScala(receiver.close()).recoverWith {
+      case eh: EventHubException if eh.getIsTransient =>
+          logInfo(s"(TID $taskId) transient failure closing a receiver, ignoring")
+          Future.successful(null);
+      case t: Throwable =>
+        t.getCause match {
+          case eh: EventHubException if eh.getIsTransient =>
+            logInfo(s"(TID $taskId) transient failure closing a receiver, ignoring")
+            Future.successful(null);
+          case _ =>
+            logInfo(s"(TID $taskId) nontransient failure closing a receiver")
+            throw t
```

Comment:
This would kill the running task.  When a retry is initiated and lands on the same executor would the receiver be marked to not be re-used?  Would we need to remove this from the cache?

Reply:
The happy path is this:
1) Spark connector gets receiver for a partition out of cache
2) Receiver is not suitable, so connector tries to close it
3) Close succeeds
4) Connector creates a new receiver
5) Spark job continues OK

The sequence of events that we were seeing in your issue is:
3) Close fails with transient error, connector goes into retry loop that accomplishes nothing and eventually throws
4) Spark job crashes

What we want instead is:
3) Close fails with transient error, connector logs the error but otherwise treats it as success
4) Connector continues on happy path and creates a new receiver
5) Spark job continues OK